src/org/sosy_lab/java_smt/solvers/apron/ApronModel.java

```diff
@@ -0,0 +1,320 @@
+      //if the node to evaluate has the variable of the current assignment, substitute the
+      // variable with the assigned value
+      if (hasVarZero) {
+        //substitutes every occurence of the variable with the value stored in model
+        Texpr1Node param1 = nodeToEvaluate.substitute(modelVar, toSubT);
+        nodeToEvaluate = param1;
+      }
+    }
+    //if the node to substitute has no variables anymore, build an equation of the form
+    // resultVar = nodeToEvaluate; This is the passed to a helper Abstract1 object; The result
+    // of the evaluation is a bound of the interval of possible values for resultVar
+    if (nodeToEvaluate.getVars().length == 0) {
+      // nodeToEvaluate - resultVar
+      Texpr1Node result = new Texpr1BinNode(Texpr1BinNode.OP_SUB, nodeToEvaluate, resultVar);
+      //nodeToEvaluate - resultVar = 0 --> resultVar = nodeToEvaluate
+      Environment environment = new Environment(new String[]{}, new String[]{resultName});
+      Tcons1 equation = new Tcons1(environment, Tcons1.EQ, result);
+      Tcons1[] c = new Tcons1[]{equation};
+      try {
+        Abstract1 abstract1 = new Abstract1(prover.getAbstract1().getCreationManager(), c);
+        //getting the lower bound of the interval which refers to all values the variable can
+        // take
+        Object bound =
+            abstract1.getBound(prover.getAbstract1().getCreationManager(), resultName).sup;
+        String strValue = bound.toString();
+        List<String> numbers = Splitter.on('/').splitToList(strValue);
+        BigInteger nominator =
+            new BigInteger(numbers.get(0));
+        ApronRatCstNode valueFormula;
+        if (numbers.size() > 1) { //for rational lower bounds
+          BigInteger denominator = new BigInteger(numbers.get(1));
+          valueFormula = new ApronRatCstNode(nominator, denominator);
+        } else { //if the value is an integer
+          valueFormula = new ApronRatCstNode(nominator, BigInteger.ONE);
+        }
+        return valueFormula;
+      } catch (ApronException e) {
+        throw new RuntimeException(e);
+      }
+    }
+    //if the nodeToEvaluate has still variables, that can not be substituted, the node is returned
+    else {
+      if (nodeToEvaluate instanceof Texpr1VarNode) {
+        return new ApronRatVarNode((Texpr1VarNode) nodeToEvaluate, this.formulaCreator);
+      } else if (nodeToEvaluate instanceof Texpr1UnNode) {
+        return new ApronRatUnaryNode((Texpr1UnNode) nodeToEvaluate);
+      } else {
+        return new ApronRatBinaryNode((Texpr1BinNode) nodeToEvaluate);
+      }
+    }
+  }
+
+  private ApronNode getIntVarValue(ApronIntVarNode varNode) {
+    ApronIntCstNode nullValue = new ApronIntCstNode(BigInteger.ZERO);
+    String varName = varNode.getVarName();
+    for (ValueAssignment assignment : model) {
+      if (varName.equals(assignment.getName())) {
+        return (ApronNode) assignment.getValueAsFormula();
+      }
+    }
+    return nullValue;
```

Comment:
It seems like you can just return and create the `new ApronIntCstNode(BigInteger.ZERO);` here. 
Also goes for the method below.